.github/workflows/tests.yaml

```diff
@@ -6,18 +6,30 @@ on:
     branches-ignore:
     - 'development/**'
 
+permissions:
+  packages: write
+  id-token: write
+  contents: read
+
 jobs:
   test:
     runs-on: ubuntu-latest
     steps:
     - name: Checkout
-      uses: actions/checkout@v2
-    - uses: actions/setup-node@v2
+      uses: actions/checkout@v4
+    - uses: actions/setup-node@v4
       with:
         node-version: '16'
+        registry-url: https://npm.pkg.github.com
     - name: install dependencies
       run: yarn install --frozen-lockfile
     - name: lint markdown
       run: yarn lint_md
     - name: lint javascript
       run: yarn lint
+    - name: Set development version
+      run: yarn version --no-git-tag-version --new-version "0.0.0-sha.${{ github.sha }}"
+    - name: Publish development version
+      run: npm publish --provenance
```

Comment:
I did not check, can we make sure this will not publish in registry with the `latest` tag. Otherwise a `npm dist-tag rm` can help remove the `latest` tag

Reply:
That's quite annoying indeed I don't like that it does publish it as latest, I'll check on that.

Reply:
I added the flag `--tag beta` into npm publish and it achieve what we want.